### DISCUSSION TITLE
How to chat in Avante without access to any of my (proprietary) code?

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [yetone/avante.nvim#950] [Q&A] [answered]
Is there a way to invoke Avante and chat with e.g. Claude's models - but without having Avante supply the LLM with any code in opened buffers or in my project etc?

In my case I'm working on proprietary code, and I need to make sure it doesn't get sent to an LLM.

Can I perhaps invoke `:AvanteAsk` with some option?

If I just invoke `:AvanteAsk`, it has pre-selected the currently opened buffer. Under "Selected files", I can remove the file with `d` but this also closes Avante.

> [!NOTE]
>
> For anyone else reading this, who might want the same feature; codecompanion.nvim does not default to sending off your code to the LLM. You have to specifically type in `#buffer <your question>` for example, for it to send the buffer contents to the LLM.

**Comment by rbhanot4739** [comment]
For me it doesn't close the chat but can continue chatting in that session, however the moment I toggle sidebar it adds the file again which can be unnecessary and unrelated context to the chat you are having.

**Comment by teto** [comment, marked as answer]
I was looking at the source to implement this but turns out it's an existing option (sad there is no `:h avante` that oculd have shown this) to achieve that:

 `lua require("avante.api").ask({ without_selection = true; })`

**Reply by Mithrandir2k18** [reply]
Can this also be used to send JUST the visual selection + virtual text in that area?
